### DISCUSSION TITLE
Issue in communication between rdf4j and vocbench3

### DISCUSSION THREAD

**Discussion by Tafehi** [eclipse-rdf4j/rdf4j#4903] [Q&A]
Hi 

I have an issue regarding rdf4j and vocbench3. Whenever I add the path of the rdf4j into vocbench and try to connect to the rdf4j I get the following error. I am using rdf4j 4.3.9 and run it on tomcat:9.0.85.

![Screenshot 2024-01-31 at 13 07 00](https://github.com/eclipse-rdf4j/rdf4j/assets/57415411/13e479b5-ba1d-4d22-b930-0cae5801885c)

Would appreciate if anyone can help me with it :)

Reagrds

Ehsan

**Comment by hmottestad** [comment]
Looks like you are creating a connection from one thread and then closing it from another. Connections aren't meant to be shared between threads and it can cause a deadlock, to mitigate this a deadlock prevention thread was started.

If you are using federation then there is a known issue with the federation code where it quite frequently will close a connection from a different thread than where it was created. 

There is another error message you might be seeing. "Interrupted thread ... but thread is still alive after 1000 ms!" means that the deadlock prevention thread did not mange to interrupt the thread that created the connection. It will typically show in the logs about 1 second after the message about closing connection from a different thread than the one that opened it.

**Comment by Tafehi** [comment]
I only enter the address of the rdf4j in vocbench and select remote repository and then create. I am also using ec2 on AWS. I do not use any federation. Actually I am not using any authentication between vocbench and rdf4j at all. Do I need one?

![Screenshot 2024-02-07 at 10 59 48](https://github.com/eclipse-rdf4j/rdf4j/assets/57415411/5f49bf58-cf3b-4edd-b7e9-11a3da8eb2f7)
